pkg/certgenerator/v1beta1/generator.go

```diff
@@ -64,26 +68,54 @@ var _ manager.Runnable = &CertGenerator{}
 var _ manager.LeaderElectionRunnable = &CertGenerator{}
 
 func (c *CertGenerator) Start(ctx context.Context) error {
-	if err := c.generate(ctx); err != nil {
+	if c.enableGenerateCerts {
+		if err := c.generate(ctx); err != nil {
+			return err
+		}
+	}
+	// We need to wait for certs to get ready even though using cert-manager.
+	klog.Info("Waiting for certs to get ready.")
+	if err := wait.ExponentialBackoffWithContext(ctx, wait.Backoff{
```

Comment:
Even though we use cert-manager, the controller must wait for certs to be injected into the Secret resource.